shadow 4.9 build error when disabled `shared`

When compiling with `--disable-shared` the build fails with:
```
[   25s] /bin/sh ../libtool  --tag=CC   --mode=link gcc  -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fpie  -pie -o lastlog lastlog.o  ../libmisc/libmisc.la ../lib/libshadow.la  -laudit -leconf 
[   25s] collect2: error: ld returned 1 exit status
[   25s] make[2]: *** [Makefile:1090: list_subid_ranges] Error 1
[   25s] make[2]: *** Waiting for unfinished jobs....
[   25s] collect2: error: ld returned 1 exit status
[   25s] make[2]: *** [Makefile:1042: get_subid_owners] Error 1
[   25s] libtool: link: gcc -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fpie -pie -o faillog faillog.o  ../libmisc/.libs/libmisc.a ../lib/.libs/libshadow.a
[   25s] libtool: link: gcc -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -fpie -pie -o lastlog lastlog.o  ../libmisc/.libs/libmisc.a ../lib/.libs/libshadow.a -laudit -leconf
[   25s] collect2: error: ld returned 1 exit status
[   25s] make[2]: *** [Makefile:1102: new_subid_range] Error 1
[   25s] collect2: error: ld returned 1 exit status
[   25s] make[2]: *** [Makefile:1038: free_subid_range] Error 1
[   25s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/shadow-4.9/src'
[   25s] make[1]: *** [Makefile:451: all-recursive] Error 1
[   25s] make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/shadow-4.9'
[   25s] make: *** [Makefile:379: all] Error 2
```

I assume this is similar to https://github.com/shadow-maint/shadow/issues/336 where the Gentoo ebuild used:
```
	--enable-shared=no
	--enable-static=yes
```

and we have:
```
AM_ENABLE_STATIC
AM_ENABLE_SHARED
```
in https://github.com/shadow-maint/shadow/blob/master/configure.ac#L25

resulting in the same condition.

System:
openSUSE Tumbleweed.
Complete compile flags:
```
  --enable-shadowgrp \                                                                                                                                        
  --enable-account-tools-setuid \                                                                                                                             
  --with-audit \                                                                                                                                              
  --with-libpam \                                                                                                                                             
  --with-sha-crypt \                                                                                                                                          
  --with-acl \                                                                                                                                                
  --with-attr \                                                                                                                                               
  --with-nscd \                                                                                                                                               
  --with-selinux \                                                                                                                                            
  --without-libcrack \                                                                                                                                        
  --disable-shared \                                                                                                                                          
  --with-group-name-max-length=32 \                                                                                                                           
  --enable-vendordir=%{_distconfdir}  
```

--- a/libsubid/api.c
+++ b/libsubid/api.c
@@ -39,8 +39,8 @@
 #include "idmapping.h"
 #include "subid.h"
 
-const char *Prog = "(libsubid)";
-FILE *shadow_logfd;
+static const char *Prog = "(libsubid)";
+static FILE *shadow_logfd;
 
 bool libsubid_init(const char *progname, FILE * logfd)
 {

